tests/unit/type/incr.tcl

```diff
@@ -182,4 +182,35 @@ start_server {tags {"incr"}} {
         r incrbyfloat foo [expr double(-1)/41]
         r get foo
     } {0}
+
+    foreach cmd {"incr" "decr" "incrby" "decrby"} {
+        set res {}
+        set expected {1 12}
+        if {[string match {*incr*} $cmd]} {
+            lappend expected 13
+        } else {
+            lappend expected 11
+        }
+
+        test {INCR[BY]/DECR[BY] operation should update encoding from raw to int} {
+            r del foo
+
+            r set foo 1
+            assert_encoding "int" foo
+            lappend res [r get foo]
+
+            r append foo 2
+            assert_encoding "raw" foo
+            lappend res [r get foo]
+
+            if {[string match {*by*} $cmd]} {
+                r $cmd foo 1
+            } else {
+                r $cmd foo
+            }
+            assert_encoding "int" foo
+            lappend res [r get foo]
+        }
+        assert_equal $res $expected
```

Comment:
this assertion is outside the `test` (that's wrong).
instead the creation of `expected` and the assertion should be moved inside the `test`.